lib/Login/SocialLogin.class.inc

```diff
@@ -1,4 +1,35 @@
 <?php
+// Autoloader for the new Facebook API
+require_once(LIB . 'External/Facebook/autoload.php');
+
+// It would be cleaner to store the facebook CSRF token in the session but this will do.
+if(!session_id()) {
+	session_start();
+}
+
+
+// Hackish but it's because we're not on v5.6 of PHP which has this builtin
+if(!function_exists('hash_equals'))
```

Comment:
I see this is also in `lib/External/Facebook/polyfills.php`, can we just load that file instead?